### PR TITLE
[+] Add ip utils

### PR DESCRIPTION
**Pull request by ArnaudValensi**

### CODE REVIEW

**Comment by ArnaudValensi** [comment]
[(4) IP Whitelist - Implement add and persist data](https://trello.com/c/XMlfmRb4/2346-4-ip-whitelist-implement-add-and-persist-data)

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+  throw new Error('Invalid rule type');
+}
+
+function contain(list, ip) {
+  return list.some((rule) => {
+    const ruleObject = stringToRuleObject(rule.value);
+
+    if (ruleObject.isValid === false) {
+      throw new Error(`Invalid rule: ${ruleObject.reason}`, rule);
```
Comment (arnaudbesnier) [inline src/index.js]: As a developer, I would not expect an error to be raised here. Why don't you `return false;` here?

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+    }
+
+    if (isIpMatchesRule(ip, ruleObject.rule)) {
+      return true;
+    }
+
+    return false;
```
Comment (arnaudbesnier) [inline src/index.js]: `return isIpMatchesRule(ip, ruleObject.rule);` is a simpler code.(1 line of code instead of 5)

Comment (arnaudbesnier) [reply src/index.js]:
```javascript
let matchFound = false;
if (ruleObject.isValid) {
  matchFound = isIpMatchesRule(ip, ruleObject.rule);
}

return matchFound;
```

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,221 @@
+const rangeCheck = require('range_check');
+const { Address6 } = require('ip-address');
+
+const IP_TYPE = 0;
+const RANGE_TYPE = 1;
+const SUBNET_TYPE = 2;
```
Comment (arnaudbesnier) [inline src/index.js]:
```javascript
const TYPE_IP = 0;
const TYPE_RANGE = 1;
const TYPE_SUBNET = 2;
```

**Pull request merged**